prism_compile.c

```diff
@@ -1834,15 +1836,15 @@ pm_compile_call(rb_iseq_t *iseq, const pm_call_node_t *call_node, LINK_ANCHOR *c
         flags |= VM_CALL_FCALL;
     }
 
-    if (rb_is_attrset_id(method_id)) {
+    if (pm_node->flags & PM_CALL_NODE_FLAGS_ATTRIBUTE_WRITE) {
         ADD_INSN1(ret, &dummy_line_node, setn, INT2FIX(orig_argc + 1));
-    }
-
-    ADD_SEND_R(ret, &dummy_line_node, method_id, INT2FIX(orig_argc), block_iseq, INT2FIX(flags), kw_arg);
-
-    if (rb_is_attrset_id(method_id)) {
+        ADD_SEND_R(ret, &dummy_line_node, method_id, INT2FIX(orig_argc), block_iseq, INT2FIX(flags), kw_arg);
         PM_POP;
     }
+    else {
+        ADD_SEND_R(ret, &dummy_line_node, method_id, INT2FIX(orig_argc), block_iseq, INT2FIX(flags), kw_arg);
+    }
```

Comment:
This is a more visible in split mode, but I felt like it was easier to read the 2 additional instructions surrounding the `method_id` into the conditional over conditionally setting the `setn` and `pop` on their own lines. But I can put this back if y'all prefer that style.

Reply:
Ehh this seems fine.